rest/rest_request_params.cc

```diff
@@ -34,7 +35,14 @@ std::string restDecode(std::string str, std::string *)
 
 Utf8String restDecode(Utf8String str, Utf8String *)
 {
-    return std::move(str);
+    CURL *curl = curl_easy_init();
+    int outlength;
+    std::unique_ptr<char, void (*)(char *)> cres(
+        curl_easy_unescape(curl, str.rawData(), str.length(), &outlength),
+        [](char *p) { curl_free(p); });
+    Utf8String utf8(cres.get(), cres.get() + outlength);
+    curl_easy_cleanup(curl);
+    return std::move(utf8);
```

Comment:
1.  I would avoid using CURL if possible.
2.  I'm sure we're not looking at the right place here.  str is *already* a UTF-8 string, isn't it?  If we're sure it must be an ASCII encoded string, it should be a std::string; the fact it's already UTF-8 tells the user it's already been decoded.

1.  I think that curl_easy_init is already called elswhere; you should be able to grep it
2.  It's thread safe
3.  The std::move in the original version is useful, but not in your version, due to compiler optimizations for returning a value that's a local on the stack.